TodoList/Src/TodoList.DI/DependencyResolvers/DependencyResolver.cs

```
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web.Http.Dependencies;
+using TodoList.Contracts.DI;
+using TodoList.DI.Containers;
+
+namespace TodoList.DI.DependencyResolvers
+{
+    public class DependencyResolver : ITodoListDependencyResolver
+    {
+        private bool _disposed;
+        internal readonly ITodoListContainer Container;
+        
+        internal DependencyResolver(ITodoListContainer todoListContainer)
+            => Container = todoListContainer;
+
+        public object GetService(Type serviceType)
+            => GetService(() => Container.Resolve(serviceType));
+
+        public IEnumerable<object> GetServices(Type serviceType)
+            => GetService(() => Container.ResolveAll(serviceType)) ?? Enumerable.Empty<object>();
+
+        private static T GetService<T>(Func<T> resolveFunc) where T : class 
+        {
+            try
+            {
+                return resolveFunc();
+            }
+            catch (DependencyResolutionFailedException)
+            {
+                return null;
```

It should throw an exception from time to time. 

Please, use `catch (DependencyResolutionFailedException exception)` and watch when and what it catches. You will (probably, I hope you will) find out, which of them it should throw and which of them not.

---

I see exceptions only with messages like this: `Resolution of the dependency failed, type = 'System.Web.Http.Metadata.ModelMetadataProvider'...`. 

I find just one difference. Almost all are `interfaces` but one is `abstract class`. Is this the problem?

---

e680234